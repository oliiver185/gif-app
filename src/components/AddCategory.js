import React, { useState } from 'react';
import PropTypes from 'prop-types'



const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');  //useState() asi es undefined
    
    const handleInputChange = (e) => {
        setInputValue (e.target.value);
    }

    const handleSubmit = (e) => {
          e.preventDefault(); 
          

        if (inputValue.trim().length > 2){
          setCategories(cats => [inputValue, ...cats ]);
          setInputValue('');
        }    

        // inputValue.trim().length > 2 && setCategories(cats => [...cats, inputValue]);
        // inputValue && setCategories( (category) => [...category, inputValue] ); 
    }

    return (       
        <form onSubmit={handleSubmit}>
          <p>{inputValue}</p>
        <input
          type="text"
          value = {inputValue}
          onChange= { handleInputChange}
          placeholder="Type something ðŸ˜€"
          /> 
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

// PrimeraApp.propTypes = {
//   saludo: PropTypes.string.isRequired
// //    otra: PropTypes.number.isRequired 
// }
export default AddCategory;