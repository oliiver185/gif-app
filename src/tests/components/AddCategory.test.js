import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import React from 'react';
import AddCategory from '../../components/AddCategory';


describe('Tests in <AddCategory />', () => {


    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach ( () => {
        jest.clearAllMocks();
         wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })
    
        test('should show correctly', () => {

            expect(wrapper).toMatchSnapshot();
        });
        
        
        test('should change the text box', () => {

            const input = wrapper.find('input');
            const value = 'hola mundo';

            input.simulate('change', {target: {
                value:value
            }});

            expect(wrapper.find('p').text().trim()).toBe(value);
            
        });

        test('should not post the information with submit', () => {
            
            wrapper.find('form').simulate('submit',  { preventDefault(){} });

            expect(setCategories).not.toHaveBeenCalled();
        });

        test('should call setCategories and clean the text box', () => {
            
            const value = 'hola';

            const input = wrapper.find('input');
            input.simulate('change',  {target: {value:value}} );

            wrapper.find('form').simulate('submit', { preventDefault(){}});

            expect(setCategories).toHaveBeenCalled();
          	
            expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
            expect(input.prop('value')).toBe('');


            
        })
        
        
        
})






