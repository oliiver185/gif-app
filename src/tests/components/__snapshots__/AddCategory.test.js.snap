// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AddCategory /> should show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Type something ðŸ˜€"
    type="text"
    value=""
  />
</form>
`;
